src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,198 @@
+        }
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        Node<T> node;
+        int nodeIndex;
+        if (index > size >> 1) {
+            node = lastNode;
+            nodeIndex = size - 1;
+            while (nodeIndex > index) {
+                node = node.getPrev();
+                nodeIndex--;
+            }
+            return node;
+        }
+        node = firstNode;
+        nodeIndex = 0;
+        while (nodeIndex < index) {
+            node = node.getNext();
+            nodeIndex++;
+        }
+        return node;
+    }
+
+    private void putNodeMiddle(T value, Node<T> operationNode) {
+        Node<T> newNode = new Node<>(value, operationNode.getPrev(), operationNode);
+        operationNode.getPrev().setNext(newNode);
+        operationNode.setPrev(newNode);
+        size++;
+    }
+
+    private void putNode(T value, String position) {
+        Node<T> newNode;
+        switch (position) {
+            case "top":
+                newNode = new Node<>(value, lastNode, null);
+                lastNode.setNext(newNode);
+                lastNode = newNode;
+                break;
+            case "bottom":
+                newNode = new Node<>(value, null, firstNode);
+                firstNode.setPrev(newNode);
+                firstNode = newNode;
+                break;
+            default:
+                newNode = new Node<>(value, null, null);
+                firstNode = newNode;
+                lastNode = newNode;
+        }
+        size++;
+    }
+
+    private void removeNode(Node<T> node) {
+        if (!node.equals(firstNode) && !node.equals(lastNode)) {
+            node.getPrev().setNext(node.getNext());
+            node.getNext().setPrev(node.getPrev());
+        }
+        if (node.equals(firstNode) && size > 1) {
+            node.getNext().setPrev(null);
+            firstNode = node.getNext();
+        }
+        if (node.equals(lastNode) && size > 1) {
+            node.getPrev().setNext(null);
+            lastNode = node.getPrev();
+        }
+        if (firstNode.equals(lastNode)) {
+            firstNode = null;
+            lastNode = null;
+        }
+        size--;
+    }
+
+    private static class Node<E> {
+        private E element;
+        private Node<E> prev;
+        private Node<E> next;
+
+        public Node(E element, Node<E> prev, Node<E> next) {
+            this.element = element;
+            this.prev = prev;
+            this.next = next;
+        }
+
+        public void setElement(E element) {
+            this.element = element;
+        }
+
+        public void setPrev(Node<E> prev) {
+            this.prev = prev;
+        }
+
+        public void setNext(Node<E> next) {
+            this.next = next;
+        }
+
+        public E getElement() {
+            return element;
+        }
+
+        public Node<E> getPrev() {
+            return prev;
+        }
+
+        public Node<E> getNext() {
+            return next;
+        }
```

Comment:
```suggestion
    private static class Node<E> {
        private E element;
        private Node<E> prev;
        private Node<E> next;

        public Node(E element, Node<E> prev, Node<E> next) {
            this.element = element;
            this.prev = prev;
            this.next = next;
        }
```
no need for getters and setters